chain/client/src/chunk_validation.rs

```diff
@@ -525,13 +534,42 @@ impl Client {
     }
 
     /// Function to process an incoming chunk endorsement from chunk validators.
+    /// We first verify the chunk endorsement and then store it in a cache.
+    /// We would later include the endorsements in the block production.
     pub fn process_chunk_endorsement(
         &mut self,
-        _endorsement: ChunkEndorsement,
+        endorsement: ChunkEndorsement,
     ) -> Result<(), Error> {
-        // TODO(10265): Here if we are the current block producer, we would store the chunk endorsement
-        // for each chunk which would later be used during block production to check whether to include the
-        // chunk or not.
+        let chunk_hash = endorsement.chunk_hash();
+        let account_id = &endorsement.account_id;
+
+        // If we have already processed this chunk endorsement, return early.
+        if self
+            .chunk_validator
+            .chunk_endorsements
+            .get(chunk_hash)
+            .is_some_and(|existing_endorsements| existing_endorsements.get(account_id).is_some())
+        {
+            tracing::debug!(target: "chunk_validation", ?endorsement, "Already received chunk endorsement.");
+            return Ok(());
+        }
+
+        let chunk_header = self.chain.get_chunk(chunk_hash)?.cloned_header();
+        if !self.epoch_manager.verify_chunk_endorsement(&chunk_header, &endorsement)? {
+            tracing::error!(target: "chunk_validation", ?endorsement, "Invalid chunk endorsement.");
+            return Err(Error::InvalidChunkEndorsement);
+        }
+
+        // If we are the current block producer, we store the chunk endorsement for each chunk which
+        // would later be used during block production to check whether to include the chunk or not.
+        tracing::debug!(target: "chunk_validation", ?endorsement, "Received and saved chunk endorsement.");
+        self.chunk_validator
+            .chunk_endorsements
+            .get_or_insert(chunk_hash.clone(), || HashMap::new());
+        let chunk_endorsements =
+            self.chunk_validator.chunk_endorsements.get_mut(chunk_hash).unwrap();
```

Comment:
I think you can use `self.chunk_validator.chunk_endorsements.entry(*chunk_hash).or_insert_with(HashMap::new).insert(account_id.clone(), endorsement);`

Reply:
Yeah, the `entry` unfortunately doesn't exist with LRU cache. I too wish it had. This is the pattern we've used in `fn on_chunk_header_ready_for_inclusion` for `prev_block_to_chunk_headers_ready_for_inclusion`